inbm/dispatcher-agent/dispatcher/dispatcher_class.py

```diff
@@ -715,15 +706,47 @@ def check_dispatcher_state_info(self) -> None:
             self._telemetry('Dispatcher detects normal boot sequence')
 
 
-def handle_updates(dispatcher: Any) -> None:
+def handle_updates(dispatcher: Any, 
+                   schedule_manifest_schema=SCHEDULE_SCHEMA_LOCATION, 
+                   manifest_schema=SCHEMA_LOCATION) -> None:
     """Global function to handle multiple requests from cloud using a FIFO queue.
 
     @param dispatcher: callback to dispatcher
     """
-    message: Tuple[str, str] = dispatcher.update_queue.get()
+    message: Tuple[str, str, Optional[str]] = dispatcher.update_queue.get()
     request_type: str = message[0]
     manifest: str = message[1]
-
+    if message[2]:
+        request_id: str = message[2]
+    
+    if request_type == "schedule":
+        logger.debug("DEBUG: manifest = " + manifest)
+        try:
+            schedule = ScheduleManifestParser(manifest, schedule_manifest_schema, manifest_schema)
+        except XmlException as e:
+            logger.error("XMLException parsing schedule: " + str(e))
+            if request_id:
+                dispatcher._send_result(f"Error parsing schedule manifest: {str(e)}", request_id)
+            return
+        
+        # TODO: Change single and repeated to add the schedules to the scheduler DB
+        if schedule.single_scheduled_requests or schedule.repeated_scheduled_requests:
+            sql_lock.acquire()
```

Comment:
When using the sqlite3 module in Python, you typically don't need to manually set locks. SQLite automatically handles locking and concurrency control internally.

(also would be good to move this to another PR)